src/main/java/core/basesyntax/strategy/impl/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationHandler;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction fruitTransaction) {
+        int initialQuantity = Storage.STORAGE.get(fruitTransaction.getFruit());
+        if (initialQuantity < fruitTransaction.getQuantity()) {
```

Comment:
Create separate  variables for fruit name and quantity to increase readability 
Apply to other handlers